libclamav/scanners.c

```diff
@@ -5629,6 +5748,9 @@ cl_error_t cli_magic_scan_file(const char *filename, cli_ctx *ctx, const char *o
     if (fd >= 0) {
         close(fd);
     }
+    // Clear the next-layer attributes so we don't accidentally apply them to subsequent layers.
+    ctx->next_layer_attributes = 0;
```

Comment:
I noticed you defined a constant for this (`LAYER_ATTRIBUTES_NORMAL`) -- might as well use it, or name it so it better reflects the cleared/initial state (e.g., `LAYER_ATTRIBUTES_NONE`).

Reply:
Good call, I like NONE